components/processor/gcomapiprocessor/config.go

```diff
@@ -0,0 +1,38 @@
+package gcomapiprocessor
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type Config struct {
+	serviceName string       `mapstructure:"service_name"`
+	Client      clientConfig `mapstructure:"client"`
+	Cache       cacheConfig  `mapstructure:"cache"`
+	IsTest      bool         `mapstructure:"test"`
```

Comment:
Dry-run? Mock? Dev? Perhaps we don't even need this if we provide a container image with a gcom API mock?

Reply:
Or perhaps we could have a special handling for an endpoint that starts with "mock://" ?

Reply:
> perhaps we don't even need this if we provide a container image with a gcom API mock?

> Or perhaps we could have a special handling for an endpoint that starts with "mock://" ?

I like both options, but would probably lean towards `mock://` since the test setup will be simpler.